File: RecallThat/Models/Memory.swift
import Foundation

/// Core domain model representing a single indexed memory (screenshot or future type).
/// All services and view models use this struct — they never touch SwiftData directly.
struct Memory: Identifiable, Equatable {
    let id: UUID
    let sourceType: MemorySourceType

    /// PHAsset localIdentifier — nil if the item did not come from Photos
    let photoAssetIdentifier: String?

    /// Path to a small local thumbnail written by ThumbnailService (Phase 6)
    var localThumbnailPath: String?

    /// When the original screenshot was captured (from PHAsset.creationDate)
    let createdAt: Date

    /// When RecallThat first indexed this memory
    let importedAt: Date

    /// Human-readable title: first meaningful OCR line, or "Screenshot from [date]"
    var title: String

    /// Full OCR text extracted by OCRService
    var ocrText: String

    var ocrStatus: OCRStatus

    /// Denormalized search index: title + ocrText, lowercased (for fast local search)
    var searchText: String

    /// True while the original Photos asset still exists in the library
    var originalExists: Bool

    /// Set when the user explicitly deletes the original via the Phase 7 flow
    var deletedOriginalAt: Date?

    /// Original URL for sharedURL memories (nil for screenshot/sharedImage/sharedText)
    var sourceURL: String?
}

File: RecallThatAndroid/app/src/main/java/com/recallthat/Models/Memory.kt
package com.recallthat.Models

import java.util.Date
import java.util.UUID

data class Memory(
    val id: UUID,
    val sourceType: MemorySourceType,
    val photoAssetUri: String?,          // content:// URI string (mirrors iOS photoAssetIdentifier)
    val localThumbnailPath: String?,
    val createdAt: Date,
    val importedAt: Date,
    val title: String,
    val ocrText: String,
    val ocrStatus: OCRStatus,
    val searchText: String,              // denormalized: (title + ocrText).lowercase()
    val originalExists: Boolean,
    val deletedOriginalAt: Date?,
    val sourceUrl: String?
)
